tools/mason/MasonSearch.chpl

```diff
@@ -46,29 +47,51 @@ proc masonSearch(args: [?d] string) {
 }
 
 proc masonSearch(ref args: list(string)) {
-  if hasOptions(args, "-h", "--help") {
+  var parser = new argumentParser();
+
+  var helpFlag = parser.addFlag("help",
+                                opts=["-h","--help"],
+                                defaultValue=false,
+                                flagInversion=false);
+  // TODO: Is the debug flag supposed to be documented in help or not?
+  var debugFlag = parser.addFlag("debug",
+                                opts=["--debug"],
+                                defaultValue=false,
+                                flagInversion=false);
+  var showFlag = parser.addFlag(name="show",
+                                opts=["--show"],
+                                defaultValue=false,
+                                flagInversion=false);
+  var updateFlag = parser.addFlag(name="update",
+                                  opts=["--update"],
+                                  flagInversion=true);
+  var queryArg = parser.addArgument(name="query",
+                                    numArgs=0..1,
+                                    defaultValue=".*");
+  try! {
+    parser.parseArgs(args.toArray());
+  }
+  catch ex : ArgumentError {
+    stderr.writeln(ex.message());
+    masonSearchHelp();
+    exit(1);
+  }
+  if helpFlag.valueAsBool() {
     masonSearchHelp();
     exit(0);
   }
 
-  const show = hasOptions(args, "--show");
-  const debug = hasOptions(args, "--debug");
+  const show = showFlag.valueAsBool();
+  const debug = debugFlag.valueAsBool();
   var skipUpdate = MASON_OFFLINE;
-  if hasOptions(args, "--update") {
-    skipUpdate = false;
-  }
-
-  if hasOptions(args, "--no-update") {
-    skipUpdate = true;
+  if updateFlag.hasValue() {
+    skipUpdate = !updateFlag.valueAsBool();
   }
 
   updateRegistry(skipUpdate);
 
-  consumeArgs(args);
-
   // If no query is provided, list all packages in registry
```

Comment:
this comment probably belongs before/in the `var queryArg = parser.addArgument(` code